### PR TITLE
Greg/navigation

### PR DESCRIPTION
**Pull request by gmferland**
This PR provides navigation and is the scaffolding for the app. I used `react-navigation` because it is the recommended package for Create React Native App. Other packages are more popular in the community and easier to use (such as `wix/react-native-navigation`), but are not compatible with our setup because they use native code.

See the [React Navigation](https://reactnavigation.org/) docs for details.

tl;dr;

* Our main navigation is called a SwitchNavigator. This combines Auth and Survey flows with the main application. This allows us to:
1. Only show the Login screen when users are not logged in
2. Only sho the Survey screen when users have not taken the survey
3. For all other users, show the main screens of the app
* The Main navigation is done through a BottomTabNavigator. This combines Home, Elections, Favorites, and Activism screens through a single bottom navigation tab. The next step in navigation is to wrap this in a DrawerNavigator to enable the Hamburger menu and left drawer seen in the mocks.
* The Auth flow, Survey flow, and each screen in the Main app are StackNavigators. Each StackNavigator is like having history in your web browser. This is where we see smooth, native, animated transitions between screens and the ability to go "forward" or "back".
* Each StackNavigator supports a variety of screens and provides the `navigation` prop to every screen used for transitions to other screens.
* Each screen is a React component containing a `View` or a similar component.

### CODE REVIEW

**Comment by mialknowles** [comment]
Looks good to me.

**Review by mialknowles — APPROVED** [review]

**Comment by 156javalos** [comment]
So does watchman automatically reloads the app?? Cause usually I just do Ctrl+s and that saves, although I have press on it a bunch of times. So maybe watchman would be better.

The only other thing I see is that we probably don't need to import the colors library? Usually I would just write lightgray or any other color as a string and that works. Also cause a lot of the colors we'll use are probably gonna be hexadecimal. Which maybe we should be consistent with either using hexadecimal or text? But that's not too much of a big deal.

Other than that it seems good to me! I'll probably be using react-navigator as well, but I need to solve a few problems on my version of it.

File: src/auth/sagas.test.js [hunk 1]
```diff
@@ -1,95 +1,114 @@
...
+      takeEvery(AuthActionType.LoginRequest, loginSaga),
+    );
+  });
 
   it('should take every logOut action and trigger logOutSaga', () => {
-    gen.next()
-    gen.next()
-    expect(gen.next().value).toEqual(takeEvery(AuthActionType.LogOutRequest, logOutSaga))
-  })
+    gen.next();
```
Comment (rahul-nath) [inline src/auth/sagas.test.js:40]: it's advisable not to skip over yields -- it's fine for now, but I think you should have a `describe` for each path flow and an `it` for each generator yield.

File: src/navigation/ActivismStack.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { createStackNavigator } from 'react-navigation';
+import ActivismScreen from '../screens/ActivismScreen';
+
+import styles from './styles';
+
+const ActivismStack = createStackNavigator(
```
Comment (rahul-nath) [inline src/navigation/ActivismStack.js:6]: I've personally only used `react-router` and thought we'd use `react-router-native`, though I'm sure this is fine.

**Review by rahul-nath — APPROVED** [review]
Looks good to me man, though I didn't run it locally.

**Review by javalos-giraldo — APPROVED** [review]
So does watchman automatically reloads the app?? Cause usually I just do Ctrl+s and that saves, although I have press on it a bunch of times. So maybe watchman would be better.

The only other thing I see is that we probably don't need to import the colors library? Usually I would just write lightgray or any other color as a string and that works. Also cause a lot of the colors we'll use are probably gonna be hexadecimal. Which maybe we should be consistent with either using hexadecimal or text? But that's not too much of a big deal.

Other than that it seems good to me! I'll probably be using react-navigator as well, but I need to solve a few problems on my version of it.

File: src/navigation/ActivismStack.js (hunk 2 above)
Comment (gmferland) [reply src/navigation/ActivismStack.js:6]: I honestly couldn't tell the difference between most of the popular navigation packages. I spent a long time trying to use `react-native-navigation` before figuring out that it's not compatible with our setup. The `react-navigation` api was easy enough for me to learn, and it seems popular and well supported on npm/github

**Comment by gmferland** [comment]
@156javalos, to address your comments:

1. I don't think Watchman is solely responsible for reloading the app in dev mode. I got an error the first time I tried to run the app, and installing Watchman was the recommended fix. On the other hand, I helped Mia set up her dev environment last night and she did not need it at all. I'm not sure if it's a mac/windows thing or if it's more sophisticated than that.

2. To me, exporting a `colors` object is a way of easily referencing our own custom colors in the app. Using the string "lightgray" or "white" or whatever maps to a very specific hexadecimal value. Even though I just picked random, close-enough values to start, my goal is to have this store a variable for every custom color picked out by our design team that we can reference everywhere else in code. That way if anything ever changed, we only have to update it in the `colors.js` file.

File: src/auth/sagas.test.js (hunk 1 above)
Comment (gmferland) [reply src/auth/sagas.test.js:40]: Generators are definitely the weak point of my Javascript knowledge. So any help in that realm is much appreciated!

**Pull request merged**